### PR TITLE
For #23771 - Set correct destructive text color for card editor error messages

### LINKED ISSUE #23771: Set the destructive text color for the credit card editor
When we display the error message from validating whether or not the entered credit card number is validate in "Add card". We aren't setting the error text color to attr/destructive. https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt#119

We should set the error text color for both `cardNumberLayout` and `nameOnCardTitle`.

### PR DESCRIPTION
**Pull request by tristan-jea**
Fixes #23771
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
Test are written in `CreditCardEditorViewTest`. All unit tests successfully ran except 3 (the 3 tests also failed without my modification).
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
No screenshots of the changes because of `SecureFragment` protecting against screenshots.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.
Not needed for privacy reasons.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt [hunk 1]
```diff
@@ -19,9 +19,7 @@ import mozilla.components.service.sync.autofill.AutofillCreditCardsAddressesStor
 import mozilla.components.service.sync.autofill.AutofillCrypto
 import mozilla.components.support.test.robolectric.testContext
 import mozilla.components.support.utils.CreditCardNetworkType
-import org.junit.Assert.assertEquals
-import org.junit.Assert.assertFalse
-import org.junit.Assert.assertTrue
+import org.junit.Assert.*
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:22]: We prefer single imports compare to wildcard imports

File: app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt [hunk 2]
```diff
@@ -291,4 +289,48 @@ class CreditCardEditorViewTest {
...
+        fragmentCreditCardEditorBinding.expiryMonthDropDown.setSelection(expiryMonth - 1)
+
+        fragmentCreditCardEditorBinding.saveButton.performClick()
+
+        verify {
+            creditCardEditorView.validateForm()
+        }
+
+        assertFalse(creditCardEditorView.validateForm())
+        assertNotNull(fragmentCreditCardEditorBinding.nameOnCardLayout.error)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:312]: I think we can simply the tests by just adding this check to `GIVEN invalid name on card WHEN the save button is clicked THEN interactor is not called` test

**Comment by gabrielluong** [comment]
ktlint:

> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:22:1: Wildcard import (cannot be auto-corrected)
> /home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt:64:93: Missing newline before ")"
> /home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt:66:93: Missing newline before ")"

**Comment by tristan-jea** [comment]
I ran the accessibility scanner but the scan returns a black screenshot (expected). Should I include it?
Screenshot_20220217_180901

**Comment by gabrielluong** [comment]
> I ran the accessibility scanner but the scan returns a black screenshot (expected). Should I include it in the PR? !

No, this is fine. We don't allow screenshots for the credit card settings for privacy reasons.

**Comment by gabrielluong** [comment]
Almost there just some more ktlint

> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:168:22: Missing newline after "("
> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:169:93: Missing newline before ")"
> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:191:22: Missing newline after "("
> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:192:93: Missing newline before ")"
> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:231:22: Missing newline after "("
> /home/runner/work/fenix/fenix/app/src/test/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorViewTest.kt:232:93: Missing newline before ")"
> /home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt:64:93: Missing newline before ")"
> /home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt:66:93: Missing newline before ")"

https://github.com/mozilla-mobile/android-components#style should be helpful

**Review by gabrielluong — APPROVED** [review]
This looks good! Thanks so much!

**Comment by gabrielluong** [comment]
Thanks again for patch! I had to manually landed this in https://github.com/mozilla-mobile/fenix/pull/23803 since mergify CI build was failing in this one.

**Comment by tristan-jea** [comment]
You’re welcome, that was a really great first experience !!

**Pull request closed without merging**